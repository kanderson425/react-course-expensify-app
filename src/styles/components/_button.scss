.button {
    border-radius: 3px;
    border: none;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
    color: white;
    display: inline-block;
    background: $blue;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}