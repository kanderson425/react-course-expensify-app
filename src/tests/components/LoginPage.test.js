import React from 'react';
import { shallow } from 'enzyme'; 
import { LoginPage } from '../../components/LoginPage';

let startLogin, history, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<LoginPage startLogin={startLogin}/>);
  });

test('should render loginPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLoogin on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});