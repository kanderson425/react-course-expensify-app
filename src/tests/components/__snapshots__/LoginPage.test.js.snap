// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loginPage correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
